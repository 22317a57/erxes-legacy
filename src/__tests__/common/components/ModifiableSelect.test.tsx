import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ModifiableSelect from '../../../modules/common/components/ModifiableSelect';

describe('Modifiable component', () => {
  const defaultProps = {
    options: [],
    value: '80',
    placeholder: 'phone',
    buttonText: 'add',
    adding: false,
    onChange: (params: { options: any[]; selectedOption: any }) => null
  };

  test('renders shallow successfully', () => {
    shallow(<ModifiableSelect {...defaultProps} />);
  });

  test('renders with default props', () => {
    const control = mount(<ModifiableSelect {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('renders mount test 2 different props', () => {
    defaultProps.buttonText = 'Add name';
    defaultProps.placeholder = 'Primary name';

    const rendered = mount(<ModifiableSelect {...defaultProps} />);
    const props = rendered.props();

    expect(defaultProps).toMatchObject(props);
  });

  test('render expect Modifiliable', () => {
    const rendered = mount(<ModifiableSelect {...defaultProps} />);
    const found = rendered.find('span').debug();
    const founded = found.search('className="Select-placeholder"');
    expect(founded).toBeGreaterThan(-1);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<ModifiableSelect {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
