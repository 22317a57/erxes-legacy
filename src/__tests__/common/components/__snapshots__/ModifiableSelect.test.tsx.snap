// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modifiable component snapshot matches 1`] = `
Array [
  <div
    className="sc-jKVCRD eDSGVr"
  >
    <div
      className="Select is-clearable Select--single"
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-5--value"
        >
          <div
            className="Select-placeholder"
          >
            Primary name
          </div>
          <div
            aria-activedescendant="react-select-5--value"
            aria-disabled="false"
            aria-expanded={false}
            aria-owns=""
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>,
  <button
    className="sc-ifAKCX eKlNl"
    icon="add"
    onClick={[Function]}
    type="button"
  >
    <i
      className="icon-add sc-bwzfXH hpxbGw"
      icon="add"
    />
    <span>
      Add name
    </span>
  </button>,
]
`;
